libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -586,22 +452,12 @@ int LinuxUARTDriver::_write_fd(const uint8_t *buf, uint16_t n)
 int LinuxUARTDriver::_read_fd(uint8_t *buf, uint16_t n)
 {
     int ret;
-    ret = ::read(_rd_fd, buf, n);
+
+    ret = _device->read(buf, n);
+
     if (ret > 0) {
         BUF_ADVANCETAIL(_readbuf, ret);
-    } else {
-        switch (errno) {
-            case EAGAIN: 
-                /* Ignore EAGAIN that resulted from non-blocking read */
-                break;
-            case EPIPE:
-                /* Ignore EPIPE that resulted from peer shutdown */
-                break;
-            default:
-               ::fprintf(stdout, "read failed - %s\n", strerror(errno));
```

Comment:
I don't think we should spam the console with these errors. If a USB serial device comes loose in flight then if we start writing these errors to the console rapidly we cause cause problems with scheduling.
We either need to close the device, or only print an error on the first failure